### PR TITLE
Add `SliderWithCallback` component

### PR DESCRIPTION
**Pull request by mikolajlubiak**
I wanted to do some actions in real time when the slider value changes, not only update the reference value, so I made a little `SliderWithCallback` component

### CODE REVIEW

**Comment by mikolajlubiak** [comment]
Example usage: https://github.com/mikolajlubiak/memory/

**Comment by ArthurSonzogni** [comment]
Thanks! This looks like a good thing to have. 

Could you please extend the current `SliderOption` instead? This would avoid duplicating the current code.

**Comment by mikolajlubiak** [comment]
@ArthurSonzogni friendly ping

**Comment by ArthurSonzogni** [comment]
Thanks! LGTM.

- I renamed `callback` into `on_change`.
- I added tests.
- I fixed bugs where `on_change` was called multiple time per change.
- I added the change to the CHANGELOG.

If anything looks wrong to you, please let me know, and we will maek the appropriate change.

File: src/ftxui/component/slider.cpp [hunk 1]
```diff
@@ -127,8 +123,11 @@ class SliderBase : public ComponentBase {
       OnUp();
     }
 
-    SetValue(util::clamp(value_(), min_(), max_()));
-    if (old_value != value_()) {
+    this->value() = std::max(this->min(), std::min(this->max(), this->value()));
```
Comment (mikolajlubiak) [inline src/ftxui/component/slider.cpp:126]: Why use this instead of std::clamp?

**Comment by mikolajlubiak** [comment]
To get to the contributors list I would have to make more contributions?

**Comment by mikolajlubiak** [comment]
Also would it be cool to add my memory game to the projects using FTXUI list or is it for more advanced projects?

**Comment by ArthurSonzogni** [comment]
> To get to the contributors list I would have to make more contributions, right?

You are in the contributor list:
image

It might take a few hours for the displayed image to be invalidated and the new version to appear.

> Also would it be cool to add my [memory game](https://github.com/mikolajlubiak/memory/) or [media (video/gif/image) into colored, animated or static, ASCII art converter](https://github.com/mikolajlubiak/terminal_animation) to the projects using FTXUI list or is it for more advanced projects?

It would be cool. Do you want to submit another PR adding it?

**Comment by mikolajlubiak** [comment]
Thanks, I made the PR: #946

`std::max(this->min(), std::min(this->max(), this->value()));`
Out of curiosity, why use this instead of `std::clamp`?

**Pull request merged**